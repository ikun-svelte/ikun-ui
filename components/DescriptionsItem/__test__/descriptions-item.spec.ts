import { tick } from 'svelte';
import { afterEach, expect, test, describe, beforeEach } from 'vitest';
import KDescriptionsItem from '../src';
import KDescriptionsItemLabel from './fixture/descriptions-item.label.svelte';

let host: HTMLElement;

const initHost = () => {
	host = document.createElement('div');
	host.setAttribute('id', 'host');
	document.body.appendChild(host);
};
beforeEach(() => {
	initHost();
});
afterEach(() => {
	host.remove();
});

describe('Test: KDescriptionsItem', () => {
	test('props: cls', async () => {
		const instance = new KDescriptionsItem({
			target: host,
			props: {
				cls: 'k-descriptions-item--test'
			}
		});
		expect(instance).toBeTruthy();
		expect((host as HTMLElement)!.innerHTML.includes('k-descriptions-item--test')).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('props: label', async () => {
		const instance = new KDescriptionsItem({
			target: host,
			props: {
				label: 'KDescriptionsItem'
			}
		});
		expect(instance).toBeTruthy();
		expect((host as HTMLElement)!.innerHTML.includes('k-descriptions-item--label')).toBeTruthy();
		const labelEl = host.querySelector('.k-descriptions-item--label');
		expect(labelEl.textContent).toBe('KDescriptionsItem');
		expect(host.innerHTML).matchSnapshot();
	});

	test('slots: label', async () => {
		const instance = new KDescriptionsItemLabel({
			target: host
		});
		expect(instance).toBeTruthy();
		await tick();
		const labelEl = host.querySelector('#slot-label');
		expect(labelEl.textContent).toBe('Ikun Descriptions Item Label');
		expect(host.innerHTML).matchSnapshot();
	});
});
