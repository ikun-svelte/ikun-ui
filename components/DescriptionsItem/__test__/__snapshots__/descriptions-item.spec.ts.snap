// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test: KDescriptionsItem > props: cls 1`] = `"<div class=\\"k-descriptions-item k-descriptions-item__dark k-descriptions-item--md k-descriptions-item__horizontal ikun-last:k-descriptions-item--col-0 k-descriptions-item--test\\"><label class=\\"k-descriptions-item--label k-descriptions-item--label__horizontal k-descriptions-item--label__horizontal--md\\"></label> <div class=\\"k-descriptions-item--content k-descriptions-item--content__horizontal\\"></div></div>"`;

exports[`Test: KDescriptionsItem > props: label 1`] = `"<div class=\\"k-descriptions-item k-descriptions-item__dark k-descriptions-item--md k-descriptions-item__horizontal ikun-last:k-descriptions-item--col-0\\"><label class=\\"k-descriptions-item--label k-descriptions-item--label__horizontal k-descriptions-item--label__horizontal--md\\">KDescriptionsItem</label> <div class=\\"k-descriptions-item--content k-descriptions-item--content__horizontal\\"></div></div>"`;

exports[`Test: KDescriptionsItem > slots: label 1`] = `"<div class=\\"k-descriptions-item k-descriptions-item__dark k-descriptions-item--md k-descriptions-item__horizontal ikun-last:k-descriptions-item--col-0\\"><label class=\\"k-descriptions-item--label k-descriptions-item--label__horizontal k-descriptions-item--label__horizontal--md\\"><div id=\\"slot-label\\" slot=\\"label\\">Ikun Descriptions Item Label</div></label> <div class=\\"k-descriptions-item--content k-descriptions-item--content__horizontal\\"></div></div>"`;
